app/Core/Router.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+namespace App\Core;
+
+class Router {
+
+    const CONTROLLERS_DIR = '/app/Controllers/';
+    const CONTROLLERS_DIR_NAMESPACE = "\\App\\Controllers\\";
+
+    private $routes;
+    private $internalRoute;
+    private $actionName;
+    private $parameters;
+
+    public function __construct() {
+        $routesPath = App::getRootPath() . '/app/Config/routes.php';
+        $this->routes = include($routesPath);
+    }
+
+    /**
+     * Returns request string
+     */
+    private function getURI() {
+        if (!empty($_SERVER['REQUEST_URI'])) {
+            return trim($_SERVER['REQUEST_URI'], '/');
+        }
+    }
+    
+    /**
+     * Reads the annotation
+     */
+    public static function annotationReader($controllerName, $actionName) {
+        $reader = new \DocBlockReader\Reader($controllerName, $actionName);
+        $method = $reader->getParameter("method");
+ 
+        if(is_array($method)) {
+            if(!in_array($_SERVER['REQUEST_METHOD'], $method)) {
+                header("HTTP/1.0 405 Method Not Allowed");
+                include_once App::getRootPath() . '/app/Config/405.php';
+                die();
+            }
+        } else {
+            if($_SERVER['REQUEST_METHOD'] != $method) {
+                header("HTTP/1.0 405 Method Not Allowed");
+                include_once App::getRootPath() . '/app/Config/405.php';
+                die();
+            }
+        }
+    }
+    
+    /**
+     * Compare the uriPattern with URI
+     */
+    private function compareUriPattern($uriPattern, $uri, $path) {
+        
+        if (preg_match("~^$uriPattern$~", $uri)) {
+
+            $controllerFullName = $this->parseRoute($uriPattern, $path, $uri);
+
+            //Create an object, call a method (Action)
+            $controllerObject = new $controllerFullName;
+            call_user_func_array(array($controllerObject, $this->actionName), $this->parameters);
```

Comment:
$result = null;
...
if() ...
$result = call_user_func_array(array($controllerObject, $this->actionName), $this->parameters);
...

return $result;